Issue 11911: Parsing and test updates

tbitonti opened the pull request:
Main PR for Epic issue #11911

tbitonti committed:
Review and enhance ManagedBeanBndTest.java.
Add missing cases to ManagedBeanBndTest.java.
Add utility methods to DDTestBase.

tbitonti committed:
Cleanup WebAppTest and WebAppTestBase.
Add partial header WebApp tests.

tbitonti committed:
Add cases for 2.2 and 2.3.

tbitonti committed:
Extend the WebApp unit tests to include 2.2, 2.3, and Jakarta cases.

tbitonti committed:
Add 1.2, 1.3, and 9.0 to JavaEEVersion.  App unit test cleanups.

tbitonti committed:
EJBJar test cleanup.

tbitonti committed:
Cleanup pass over EJB unit tests.

tbitonti committed:
More EJB unit test cleanups.

tbitonti committed:
Application and application client unit test cleanups.

tbitonti committed:
Misc unit test cleanups.

tbitonti committed:
Replicating DDParser changes to all subtypes.

tbitonti committed:
First pass getting unit tests to run.

tbitonti committed:
Turn WebAppHeader tests; fix selection to correctly use the max schema version instead of the max platform version.

tbitonti committed:
Put in error message codes for Permissions unit tests.

tbitonti committed:
Fix permissions unit tests.  Add version strings to JavaEEVersion.

tbitonti committed:
First pass over FacesConfig tests.
Added getVersion test for all faces config version combinations.

tbitonti committed:
Add JSF header tests for partial header cases.

tbitonti committed:
Add EJB header tests.
Adjust 'EJBJarType.finish' to handle cases of mission versions.

tbitonti committed:
Adjusted 'finish' to correctly generate XML text.
Added application tests.

tbitonti committed:
Add alternate messages for running in eclipse.
Tune some test case.

tbitonti committed:
Convert AppBnd.  Add header tests for App and AppBnd.

tbitonti committed:
Completed conversions for BND and EXT.  (WebServicesDD and WsClientBnd, and
ValidationConfig are incomplete.
Added subtypes DDParserBndExt and DDParserSpec to DDParser, and refactored DDParser across this division.

tbitonti committed:
Updates to parsers complete except for WebServicesDDParser.

tbitonti committed:
Refactor root element name checking.

tbitonti committed:
Remove redundant 'platformVersion' from DDParser.
Unify 'finish' implementations.
Unify version to-string conversions.

tbitonti committed:
Midway through test refactoring.

tbitonti committed:
Fixing EJBBND tests.

tbitonti committed:
Completed refactoring tests.  Debugging ValidationConfig test.

tbitonti committed:
Enable multiple cache values to the test path.  Add more complete module information to the test cache data.  Get EJB and Web BND and EXT tests running again.  Relax parameterization of unit test class keyed data.

tbitonti committed:
Completed EJB BND and EXT unit tests.
Jut started updating WEB, WEB BND, and WEB EXT.

tbitonti committed:
Completed pass through unit tests.  Header tests are complete.

tbitonti committed:
Updating DD adapters.

tbitonti committed:
Fix expected error message text.

tbitonti committed:
Fold in updated messages.

tbitonti committed:
Correct unit tests to use the new messages.

tbitonti commented:
#build

helyarp commented on dev/com.ibm.ws.javaee.ddmodel/resources/com/ibm/ws/javaee/internal/ddmodel/resources/DDModelMessages.nlsprops:
```diff
@@ -34,9 +34,22 @@ required.attribute.missing=CWWKC2251E: The {2} element is missing the required {
+#    {2} is the actual root element name
+#    {3} is the expected root element name
+# -----------------------------------------------------------------------
+unexpected.root.element=CWWKC2252E: The deployment descriptor {0}, at line {1}, has root element {2}, but requires root element {3}. 
```
Change "deployment descriptor {0}" to "{0} deployment descriptor"

tbitonti replied:
Updated.

helyarp commented on dev/com.ibm.ws.javaee.ddmodel/resources/com/ibm/ws/javaee/internal/ddmodel/resources/DDModelMessages.nlsprops:
```diff
@@ -70,21 +83,94 @@ invalid.href.prefix=CWWKC2260E: The href attribute value of the {2} element does
+#    {1} is the number of a line within the deployment descriptor
+#    {2} is a version attribute value within the deployment descriptor
+# -----------------------------------------------------------------------
+unsupported.descriptor.version=CWWKC2261E: The deployment descriptor {0}, at line {1}, specifies unsupported version {2}.
```
Change "deployment descriptor {0}" to "{0} deployment descriptor"

tbitonti replied:
Updated.

helyarp commented on dev/com.ibm.ws.javaee.ddmodel/resources/com/ibm/ws/javaee/internal/ddmodel/resources/DDModelMessages.nlsprops:
```diff
@@ -70,21 +83,94 @@ invalid.href.prefix=CWWKC2260E: The href attribute value of the {2} element does
+#    {1} is the number of a line within the deployment descriptor
+#    {2} is a namespace value within the deployment descriptor
+# -----------------------------------------------------------------------
+unsupported.descriptor.namespace=CWWKC2262E: The deployment descriptor {0}, at line {1}, specifies unsupported namespace {2}.
```
Change "deployment descriptor {0}" to "{0} deployment descriptor"

tbitonti replied:
Updated.

helyarp commented on dev/com.ibm.ws.javaee.ddmodel/resources/com/ibm/ws/javaee/internal/ddmodel/resources/DDModelMessages.nlsprops:
```diff
@@ -70,21 +83,94 @@ invalid.href.prefix=CWWKC2260E: The href attribute value of the {2} element does
+#    {1} is the number of a line within the deployment descriptor
+#    {2} is a namespace value within the deployment descriptor
+# -----------------------------------------------------------------------
+unsupported.descriptor.namespace=CWWCK27780E: The deployment descriptor {0}, at line {1}, specifies unsupported namespace {2}.
```
Change "deployment descriptor {0}" to "{0} deployment descriptor"

tbitonti replied:
Updated.

helyarp commented on dev/com.ibm.ws.javaee.ddmodel/resources/com/ibm/ws/javaee/internal/ddmodel/resources/DDModelMessages.nlsprops:
```diff
@@ -70,21 +83,94 @@ invalid.href.prefix=CWWKC2260E: The href attribute value of the {2} element does
+#    {3} is a namespace value within the deployment descriptor
+#    {4} is the correct namespace value which should be present in the deployment descriptor
+# -----------------------------------------------------------------------
+unsupported.descriptor.namespace.for.version=CWWCK27781W: The deployment descriptor {0}, at line {1}, specifies version {2} and namespace {3}, but should have namespace {4}.
```
Change "deployment descriptor {0}" to "{0} deployment descriptor"

tbitonti replied:
Updated.

helyarp commented on dev/com.ibm.ws.javaee.ddmodel/resources/com/ibm/ws/javaee/internal/ddmodel/resources/DDModelMessages.nlsprops:
```diff
@@ -70,21 +83,94 @@ invalid.href.prefix=CWWKC2260E: The href attribute value of the {2} element does
+#    {2} is a version attribute value within the deployment descriptor
+#    {3} is the highest version version attribute value which is supported by the features currently provisioned by the server configuration
+# -----------------------------------------------------------------------
+unprovisioned.descriptor.version=CWWKC2263E: The deployment descriptor {0}, at line {1}, specifies version {2}, which is higher than the current provisioned version {3}.
```
Change "deployment descriptor {0}" to "{0} deployment descriptor"

tbitonti replied:
Updated.

helyarp commented on dev/com.ibm.ws.javaee.ddmodel/resources/com/ibm/ws/javaee/internal/ddmodel/resources/DDModelMessages.nlsprops:
```diff
@@ -70,21 +83,94 @@ invalid.href.prefix=CWWKC2260E: The href attribute value of the {2} element does
+#    {1} is the number of a line within the deployment descriptor
+#    {2} is a namespace value which is required to be present in the deployment descriptor
+# -----------------------------------------------------------------------
+missing.descriptor.namespace=CWWKC2264E: The deployment descriptor {0}, at line {1}, does not have required namespace {2}.
```
Change "deployment descriptor {0}" to "{0} deployment descriptor"

tbitonti replied:
Updated.

helyarp commented on dev/com.ibm.ws.javaee.ddmodel/resources/com/ibm/ws/javaee/internal/ddmodel/resources/DDModelMessages.nlsprops:
```diff
@@ -110,13 +196,23 @@ found.duplicate.attribute.value=CWWKC2270E: The {2} attribute for all {1} elemen
-BVKEY_NOT_A_BEAN_VALIDATION_XML=CWWKC2271W: A validation.xml file was found for the {0} module. However, that validation.xml file is not configured for validation; therefore this XML file is ignored.
-BVKEY_NOT_A_BEAN_VALIDATION_XML.explanation=The BeanValidationService cannot create a ValidatorFactory because of an error parsing the validation.xml file with the validation-configuration-1.0.xsd schema file. This error may indicate that the validation.xml file is not intended for use by the Bean Validation run-time; therefore, this XML file is ignored.
-BVKEY_NOT_A_BEAN_VALIDATION_XML.useraction=If the validation.xml file is not for creating a ValidatorFactory instance, then ignore the message. However if the validation.xml file is intended to create a ValidatorFactory instance, then a problem exists.  Enable Bean Validation run-time trace in the application server, stop the application, restart the application, and examine the trace output file ...
+unexpected.root.element.bval=CWWKC2271W: Ignoring bean validation configuration file {0} of module {1}.  At line number {2}, the root element is {3} but should be {4}.
```
Change "Ignoring bean validation configuration file {0} of module {1}. At line number {2}, the root element is {3} but should be {4}"
to:
"Ignoring the {0} bean validation configuration file of the {1} module. At line number {2}, replace the root element {3} with {4}."

helyarp commented on dev/com.ibm.ws.javaee.ddmodel/resources/com/ibm/ws/javaee/internal/ddmodel/resources/DDModelMessages.nlsprops:
```diff
@@ -110,13 +196,23 @@ found.duplicate.attribute.value=CWWKC2270E: The {2} attribute for all {1} elemen
-BVKEY_NOT_A_BEAN_VALIDATION_XML.useraction=If the validation.xml file is not for creating a ValidatorFactory instance, then ignore the message. However if the validation.xml file is intended to create a ValidatorFactory instance, then a problem exists.  Enable Bean Validation run-time trace in the application server, stop the application, restart the application, and examine the trace output file ...
+unexpected.root.element.bval=CWWKC2271W: Ignoring bean validation configuration file {0} of module {1}.  At line number {2}, the root element is {3} but should be {4}.
+unexpected.root.element.bval.explanation=The validation configuration file does not match the validation configuration schemas (validation-configuration-1.0.xsd and validation-configuration-1.1.xsd). Either the configuration file was not intended for use for bean validation, or the configuration file is malformed.  The configuration file is being ignored.  The BeanValidationService will not creat ...
+unexpected.root.element.bval.useraction=If the validation configuration file is not intended for bean validation, the warning message can be ignored.  If the validation configuration file is intended for bean validation, the warning message should not be ignored, as the bean validation service is not running as intended.  To diagnose the problem, enable bean validation runtime trace, stop and res ...
```
Change "the warning message should not be ignored, as" to "do not ignore the warning message because"

helyarp commented on dev/com.ibm.ws.javaee.ddmodel/resources/com/ibm/ws/javaee/internal/ddmodel/resources/DDModelMessages.nlsprops:
```diff
@@ -141,3 +237,23 @@ module.name.not.specified.useraction=Add a moduleName attribute to the web-bnd,
+#    {1} is the number of a line within the deployment descriptor
+#    {2} is a DTD public ID value which is present in the descriptor
+# -----------------------------------------------------------------------
+unsupported.descriptor.public.id=CWWCK27782E: The deployment descriptor {0}, at line {1}, specifies unsupported public ID {2}.
```
Change "deployment descriptor {0}" to "{0} deployment descriptor"

Please do a global search and replace for this phrase.

helyarp commented on dev/com.ibm.ws.javaee.ddmodel/resources/com/ibm/ws/javaee/internal/ddmodel/resources/DDModelMessages.nlsprops:
```diff
@@ -141,3 +237,23 @@ module.name.not.specified.useraction=Add a moduleName attribute to the web-bnd,
+#    {2} is a DTD public ID value which is present in the descriptor
+# -----------------------------------------------------------------------
+unsupported.descriptor.public.id=CWWCK27782E: The deployment descriptor {0}, at line {1}, specifies unsupported public ID {2}.
+unsupported.descriptor.public.id.explanation=The public ID value which is present is not valid for the deployment descriptor.
```
Remove "which is present"

helyarp commented on dev/com.ibm.ws.javaee.ddmodel/resources/com/ibm/ws/javaee/internal/ddmodel/resources/DDModelMessages.nlsprops:
```diff
@@ -141,3 +237,23 @@ module.name.not.specified.useraction=Add a moduleName attribute to the web-bnd,
+# PARAMETERS:
+#    {0} is the path to a deployment descriptor, including paths to parent archives
+# -----------------------------------------------------------------------
+missing.descriptor.version=CWWCK27783E: The deployment descriptor {0} specifies neither a version, nor a PUBLIC ID, nor a schema.
```
Change "The deployment descriptor {0} specifies neither a version, nor a PUBLIC ID, nor a schema."
to:
"The {0} deployment descriptor does not specify a version, PUBLIC ID, or schema."

tbitonti replied:
Missed this one ... updating.

tbitonti replied:
Updated.

helyarp reviewed (commented):
Suggested changes.

skasund commented:
L2 message review completed

tbitonti committed:
Extend DDModel FAT tests: Add partial incorrect header cases and add minimal header cases.

tbitonti committed:
Added bodies to many unit tests.  Added more checking to bindings and extensions reference resolution.

tbitonti committed:
Finished enabling ddmodel_fat partial and minimal test variations.

tbitonti commented:
#build

tbitonti committed:
Debugged ddmodel.ws unit tests.

tbitonti committed:
Correct messages per review.

tbitonti committed:
Correct unit test error message code for removed duplicate message.

tbitonti committed:
Additional unit test correction for removed duplicate error message.

helyarp commented on dev/com.ibm.ws.javaee.ddmodel/resources/com/ibm/ws/javaee/internal/ddmodel/resources/DDModelMessages.nlsprops:
```diff
@@ -269,7 +258,7 @@ found.duplicate.attribute.value.useraction=Correct the deployment descriptor by
 # BVKEY_NOT_A_BEAN_VALIDATION_XML=CWWKC2271W: A validation.xml file was found for the {0} module. However, that validation.xml file is not configured for validation; therefore this XML file is ignored.
 # BVKEY_NOT_A_BEAN_VALIDATION_XML.explanation=The BeanValidationService cannot create a ValidatorFactory because of an error parsing the validation.xml file with the validation-configuration-1.0.xsd schema file. This error may indicate that the validation.xml file is not intended for use by the Bean Validation run-time; therefore, this XML file is ignored.
-# BVKEY_NOT_A_BEAN_VALIDATION_XML.useraction=If the validation.xml file is not for creating a ValidatorFactory instance, then ignore the message. However if the validation.xml file is intended to create a ValidatorFactory instance, then a problem exists.  Enable Bean Validation run-time trace in the application server, stop the application, restart the application, and examine the trace output fi ...
+# BVKEY_NOT_A_BEAN_VALIDATION_XML.useraction=If the validation.xml file is not for creating a ValidatorFactory instance, then ignore the message. However if the validation.xml file is intended to create a ValidatorFactory instance, then a problem exists. Enable Bean Validation run-time trace in the application server, stop the application, restart the application, and examine the trace output fil ...
```
Change "run-time" to "runtime"

tbitonti replied:
Updated.

helyarp reviewed (commented):
Wow! Nice changes. Just one more suggestion: Change "run-time" to "runtime"

tbitonti committed:
Change "run-time" to "runtime", per updated message standards.

tbitonti committed:
Correct "missing.descriptor.version" message text.

helyarp reviewed (commented):
Looks good!

tbitonti committed:
Bindings and extension adapter refactoring.

tbitonti committed:
Extend BndExt adapter update to WebservicesBndAdapter.  Cleanup BndExt updates.

tbitonti committed:
Merge branch 'integration' into dd-test-additions

tbitonti committed:
Update and get working the Bnd and Ext tests, in particular, the non-valid cases.

tbitonti commented:
#build

tbitonti commented:
#build

tbitonti committed:
Minor version increase to web and appbnd.

tbitonti committed:
Remove faulty 'MODULE_NAME_NOT_FOUND' check.

tbitonti commented:
#build

tbitonti committed:
Turn down CromComponentReferenceType warnings to match the prior cases.  Adding new warnings causes a lot of test failures.

tbitonti commented:
#build

tbitonti committed:
Correct EJB reference in 'testCMPConnectionFactory'.

tbitonti commented:
#build

tbitonti committed:
Correct cross-component warning and error handling per issue 19207.

tbitonti committed:
Comment out "not found" case in CrossComponentReferenceType.  Enablement will be through issue 19207.  Emitting the warning causes a couple of dozen FAT test failures.
Correct version value in WebAppDDParser, which assigned version 2.2 for a 2.3 descriptor.

tbitonti commented:
#build

tbitonti commented:
All failures are from the now obsolete "com.ibm.ws.javaee.ddmodel_fat" in Commercial Liberty:

```
com.ibm.ws.javaee.ddmodel_fat FAT tests
  com.ibm.ws.javaee.ddmodel.suite.ErrorTests
    testInvalidEJBBndModuleName
    testInvalidWebBndModuleName
    testInvalidEJBExtModuleName
    testInvalidWebExtModuleName
```

The obsolete PR are being removed by Commercial Liberty pull request https://github.ibm.com/websphere/WS-CD-Open/pull/25339.

tbitonti commented:
#libby

tbitonti commented:
Testing per https://github.com/OpenLiberty/open-liberty/issues/19274
